test/csuite/wt8057_compact_stress/main.c

```diff
@@ -0,0 +1,381 @@
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+#include <sys/wait.h>
+#include <signal.h>
+
+/*
+ * This test verifies that there are no data inconsistencies if compact operation is interrupted by
+ * an unclean shutdown. To achieve this, the main process spawns a child process. The child process
+ * perform certain operations on two identical tables. The parent process randomly kills child
+ * process and verifies that the data across two tables match after restart.
+ */
+
+#define NUM_RECORDS 100000
+#define TIMEOUT 40
+
+/* Constants and variables declaration. */
+/*
+ * You may want to add "verbose=[compact,compact_progress]" to the connection config string to get
+ * better view on what is happening.
+ */
+static const char conn_config[] = "create,cache_size=2GB,statistics=(all)";
+static const char table_config[] =
+  "allocation_size=4KB,leaf_page_max=4KB,key_format=i,value_format=S";
+static char data_str[1024] = "";
+
+static const char ckpt_file[] = "WT_TEST.compact-stress/checkpoint_done";
+static const char working_dir[] = "WT_TEST.compact-stress";
+static const char uri1[] = "table:compact1";
+static const char uri2[] = "table:compact2";
+
+/*
+ * subtest_error_handler --
+ *     Error event handler.
+ */
+static int
+subtest_error_handler(
+  WT_EVENT_HANDLER *handler, WT_SESSION *session, int error, const char *message)
+{
+    (void)(handler);
+    (void)(session);
+    (void)(error);
+    fprintf(stderr, "%s", message);
+    return (0);
+}
+
+static WT_EVENT_HANDLER event_handler = {
+  subtest_error_handler, NULL, /* Message handler */
+  NULL,                        /* Progress handler */
+  NULL                         /* Close handler */
+};
+
+static void sig_handler(int) WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+
+/* Forward declarations. */
+static void run_test(const char *home);
+static void populate(WT_SESSION *session, int start, int end);
+static void remove_records(WT_SESSION *session, const char *uri, int start, int end);
+static void verify_tables(WT_SESSION *session);
+static uint64_t get_file_size(WT_SESSION *session, const char *uri);
+
+/*
+ * Signal handler to catch if the child died unexpectedly.
+ */
+static void
+sig_handler(int sig)
+{
+    pid_t pid;
+
+    WT_UNUSED(sig);
+    pid = wait(NULL);
+    /*
+     * The core file will indicate why the child exited. Choose EINVAL here.
+     */
+    testutil_die(EINVAL, "Child process %" PRIu64 " abnormally exited", (uint64_t)pid);
+}
+
+/* Methods implementation. */
+int
+main(int argc, char *argv[])
+{
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+
+    struct sigaction sa;
+    struct stat sb;
+    TEST_OPTS *opts, _opts;
+    char home[1024];
+    int status;
```

Comment:
Do we need to sort variable declarations alphabetically?

Reply:
I have rearranged the declarations in 823dc67.